### PR TITLE
go package checks

### PR DESCRIPTION
**Pull request by akaspin**
[trello](https://trello.com/c/tmyMQJxq/80-create-reusable-workflow-for-golang-lint-and-test)

### CODE REVIEW

File: .github/workflows/go-checks.yml [hunk 1]
```diff
@@ -24,21 +33,15 @@ jobs:
...
         GH_AUTH_TOKEN: ${{ secrets.gh_auth_token }}
       run: |
         git config --global url."https://${GH_AUTH_TOKEN}@github.com/".insteadOf https://github.com/
 
-    - name: Test
-      env:
-        GOPRIVATE: github.com/remerge
-      run: make lint test-nocache
+    - run: make lint
+    - run: make test-nocache
```
[the remark this answers is not in the document]
Comment (akaspin) [reply .github/workflows/go-checks.yml:46]: It's a good question. Actually linting, tests and race checks should be configured in Go sources and root Makefile. Can you kick me to specific repos?

[the remark this answers is not in the document]
Comment (tundeaoni) [reply .github/workflows/go-checks.yml:46]: @akaspin  https://github.com/remerge/act/blob/master/.circleci/config.yml#L31-L34

[the remark this answers is not in the document]
Comment (akaspin) [reply .github/workflows/go-checks.yml:46]: Brutal. It's caused by outdated test system (`... race: limit on 8128 simultaneously alive goroutines is exceeded, dying`). Yep. It's worth to make race target configurable.

[the remark this answers is not in the document]
Comment (akaspin) [reply .github/workflows/go-checks.yml:50]: done. added `no-race` input.

**Review by tundeaoni — APPROVED** [review]

**Pull request merged**